File: fapi/src/main/java/stock_exchange/service/BrokerService.java
package stock_exchange.service;
import stock_exchange.model.Broker;
import stock_exchange.model.BrokerStatisticsDTO;
import stock_exchange.model.StockExchange;
import stock_exchange.model.UnemployedBroker;
import stock_exchange.model.response.MessageResponse;
import stock_exchange.model.response.PageResponse;

import java.util.List;

public interface BrokerService {
   PageResponse<UnemployedBroker> findAllUnemployed(String title, int page, int size, String sort);

    PageResponse<Broker> findBrokers(int page, int size, int clientId);

    List<Broker> findBrokers( int clientId);

    MessageResponse employ(int brokerId, int clientId);

    MessageResponse dismiss(int brokerId);

    PageResponse<BrokerStatisticsDTO> getStatistics(int page, int size, int clientId);

    MessageResponse update(int brokerId, String exchange);
}

File: frontend/src/app/services/broker/broker.service.ts
import {HttpClient, HttpParams} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {UnemployedBroker} from '../../model/unemployed-broker';
import {Broker} from '../../model/broker';
import {BrokerStatistics} from "../../model/BrokerStatistics";
import {StockExchange} from "../../model/StockExchange";

const baseUrl = 'http://localhost:8080/broker/';

@Injectable({
  providedIn: 'root'
})
export class BrokerService {

  constructor(private http: HttpClient) {
  }

    getAll(title: string, page: number, size: number, sort: string[]): Observable<any> {
    return this.http.get(baseUrl + 'find/unemployed' + `?title=${title}&page=${page}&size=${size}&sort=${sort}`);
  }

  getClientsBrokers(page: number, size: number, clientId: number): Observable<any> {
    return this.http.get(baseUrl + 'find/all' + `?page=${page}&size=${size}&client-id=${clientId}`);
  }

  getBrokers(clientId: number): Observable<any> {
    return this.http.get(baseUrl + 'find' + `?client-id=${clientId}`);
  }

  employ(brokerId: number, clientId: number): Observable<any> {
    return this.http.post(baseUrl + 'employ/' + brokerId + '/' + clientId, null);
  }

  dismiss(brokerId: number): Observable<any> {
    return this.http.post(baseUrl + 'dismiss/' + brokerId, null);
  }

  getStatistics(page: number, size: number, clientId: number): Observable<any> {
    return this.http.get(baseUrl + 'find/statistics' + `?page=${page}&size=${size}&clientId=${clientId}`);

  }

  update(id: number, exchange: string): Observable<any> {
    return this.http.post(baseUrl + 'update/' + id + '/' + exchange, null);
  }
}
